server/polar/order/service.py

```diff
@@ -412,33 +412,16 @@ async def create_order_from_stripe(
         # Get or create customer user
         assert invoice.customer is not None
         stripe_customer_id = get_expandable_id(invoice.customer)
-        if user is None:
-            user = await user_service.get_by_stripe_customer_id(
+        if customer is None:
+            customer = await customer_service.get_by_stripe_customer_id(
                 session, stripe_customer_id
             )
-            if user is None:
-                assert invoice.customer_email is not None
-                signup_attribution = UserSignupAttribution(
-                    intent="purchase",
-                    order=order.id,
-                )
-                if order.subscription:
-                    signup_attribution = UserSignupAttribution(
-                        intent="subscription",
-                        subscription=order.subscription.id,
-                    )
-
-                user, _ = await user_service.get_by_email_or_create(
-                    session,
-                    invoice.customer_email,
-                    signup_attribution=signup_attribution,
+            if customer is None:
+                customer = await customer_service.create_from_stripe_invoice(
+                    session, invoice, product.organization
```

Comment:
We already see with changes like this how the implementation feels more natural and straightforward.

I usually tend to think that if the implementation looks good, it's a good sign that the underlying design choice and architecture is right 😄